uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/ContactWorkPatternTest.php

```diff
@@ -126,4 +129,47 @@ public function testGetForDateReturnsNullIfTheEmployeeHasWorkPatternsButTheGiven
     $this->assertNull($contactWorkPattern);
   }
 
+  public function testGetWorkPatternForContact() {
+    //create a contact with valid contract and a Work pattern assigned
+    $contact = ContactFabricator::fabricate();
+    $periodStartDate = date('2016-01-01');
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => $contact['id']
+    ],
+    [
+      'period_start_date' => $periodStartDate,
+    ]);
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'pattern_id' => $workPattern->id
+    ]);
+    $date= new DateTime('2016-01-20');
+
+    $returnedWorkPattern = ContactWorkPattern::getWorkPattern($contact['id'], $date);
+    $this->assertEquals($workPattern->id, $returnedWorkPattern->id);
+  }
+
+  public function testGetContactWorkPatternStartDate() {
+    //create a contact with valid contract and a Work pattern assigned
+    $contact = ContactFabricator::fabricate();
+    $periodStartDate = new DateTime('2016-01-01');
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => $contact['id']
+    ],
+    [
+      'period_start_date' => $periodStartDate->format('Y-m-d'),
+    ]);
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'pattern_id' => $workPattern->id
+    ]);
+    $date= new DateTime('2016-01-20');
+
+    $returnedStartDate = ContactWorkPattern::getStartDate($contact['id'], $date);
+    $this->assertEquals($periodStartDate, $returnedStartDate);
+  }
```

Comment:
And we need some more tests here too:
- Test that it returns the right date when the contact has multiple work patterns
- Test that it return the right date when the contact has no work pattern assigned, but has a contract overlapping the given date
- Test that is return the right date (or null) when the contact has no work pattern assigned and no contract overlapping the given date